plugins/json_importer.cc

```diff
@@ -68,14 +71,46 @@ JSONImporterConfigDescription()
     addField("limit", &JSONImporterConfig::limit,
              "Maximum number of lines to process");
     addField("offset", &JSONImporterConfig::offset,
-            "Skip the first n lines.", int64_t(0));
+             "Skip the first n lines.", int64_t(0));
     addField("ignoreBadLines", &JSONImporterConfig::ignoreBadLines,
              "If true, any line causing an error will be skipped. Any line "
              "with an invalid JSON object will cause an error.", false);
+    addField("where", &JSONImporterConfig::where,
+             "Which lines to use to create rows.",
+             SqlExpression::TRUE);
 
     addParent<ProcedureConfig>();
 }
 
+struct JsonScope : SqlExpressionMldbScope {
+    vector<tuple<PathElement, ExpressionValue>> row;
```

Comment:
You should keep a const reference to the ExpressionValue, and use getNestedColumn(), rather than destructure it (getStructured() is not guaranteed to work and you're making a copy of each row which is highly inefficient).